x-pack/plugins/security_solution/server/plugin.ts

```diff
@@ -532,7 +532,7 @@ export class Plugin implements ISecuritySolutionPlugin {
     );
 
     plugins.fleet?.fleetSetupCompleted().then(() => {
-      if (plugins.taskManager) {
+      if (plugins.taskManager && process.env.BUILDKITE !== 'true') {
```

Comment:
End goal is good here, we don't need the watchdog running in the test environment (esp. if there are tests verifying watchdog behavior).

But good practice is to keep all knowledge of tests and test environments outside of the "production side" of the code. Probably a few ways to achieve this, but see if you can reorganize so you can control this from a `.test.ts` file.

Could either try to hook the plugin start earlier/somewhere else and overwrite/inject the `checkMetadataTransformTask` to be a mock or no-op so this code block still runs and does nothing.

Or if thats not feasible, could try to make a global or imported variable you can replace this check with like

```
if (plugins.taskManager && RUN_WATCHDOG)
```

and by default thats just set to `RUN_WATCHDOG=true` somewhere in production code. And from the test side, give yourself the ability to mock the import or otherwise set to false.

Reply:
great point. since this is blocking all of our FTR tests and the retry logic alone is sufficient for restoring our tests, I'll remove this logic from this PR and work on it as a separate PR.